hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -422,6 +423,15 @@ private boolean shouldRedirect(FSOperationType type, TracingContext context)
   @Override
   public FSDataOutputStream create(final Path f, final FsPermission permission, final boolean overwrite, final int bufferSize,
       final short replication, final long blockSize, final Progressable progress) throws IOException {
+    return create(f, permission, overwrite, bufferSize, replication, blockSize,
```

Comment:
Need for additional parameter ?

Reply:
This method is called from `createNonRecursive` which already checks the parent dir and acquire lease on the parent dir if its an atomicDirectory. The new field will tell the method that the parent dir exists and has been checked.

Why we dont want this method to check parentDir again?
1. It has already been checked.
2. Since the dir is leased. GetPathStatus on the parentDir will fail if leaseId is not present in the API call.

Reply:
Have added javadoc in the new method.